File: Java-version/LeetCode/src/com/jooc/GazeOffer/Solution_25.java
package com.jooc.GazeOffer;

import com.jooc.components.ListNode;

public class Solution_25 {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode res = new ListNode(-1);
        ListNode cur = res;

        while(l1 != null && l2!= null){
            if(l1.val <= l2.val){
                cur.next = new ListNode(l1.val);
                l1 = l1.next;
            }else{
                cur.next = new ListNode(l2.val);
                l2 = l2.next;
            }
            cur = cur.next;
        }

        cur.next = l1 == null ? l2 : l1;

        return res.next;
    }
}

File: Swift-version/LeetCode/LeetCode/Solutions/GazeOffer/Solution_25.swift
//
//  Solution_24.swift
//  LeetCode
//
//  Created by 齐旭晨 on 2021/3/30.
//

import Foundation

class Solution_25: Solution {
    func main(){
        
    }
    
    func mergeTwoLists(_ l1: ListNode?, _ l2: ListNode?) -> ListNode? {
        var l1 = l1, l2 = l2
        let head = ListNode(-1)
        var cur = head
        
        while(l1 != nil && l2 != nil){
            if l1!.val <= l2!.val{
                cur.next = ListNode(l1!.val)
                l1 = l1?.next
            }else{
                cur.next = ListNode(l2!.val)
                l2 = l2?.next
            }
            cur = cur.next!
        }
        while l1 != nil{
            cur.next = ListNode(l1!.val)
            l1 = l1!.next
            cur = cur.next!
        }
        while l2 != nil{
            cur.next = ListNode(l2!.val)
            l2 = l2!.next
            cur = cur.next!
        }
        return head.next
    }
}
